x/ccv/provider/keeper/double_vote.go

```diff
@@ -0,0 +1,109 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerDoubleVoting verifies a double voting evidence for a given a consumer chain and,
+// if successful, executes the the jailing and the tombstoning of the malicious validator.
+// Note that the infraction header argument is temporarily only used to get the consumer chain ID.
+func (k Keeper) HandleConsumerDoubleVoting(ctx sdk.Context, evidence *tmtypes.DuplicateVoteEvidence, h *ibctmtypes.Header) error {
+	chainID := h.Header.ChainID
+
+	// get the validator's consensus address on the provider
+	providerAddr := k.GetProviderAddrFromConsumerAddr(
+		ctx,
+		chainID,
+		types.NewConsumerConsAddress(sdk.ConsAddress(evidence.VoteA.ValidatorAddress.Bytes())),
+	)
+
+	// get the consumer chain public key assigned to the validator
+	consuPubkey, ok := k.GetValidatorConsumerPubKey(ctx, chainID, providerAddr)
+	if !ok {
+		return fmt.Errorf("cannot find public key for validator %s and consumer chain %s", providerAddr.String(), chainID)
+	}
+
+	// verifies the double voting evidence using the consumer chain public key
+	if err := k.VerifyDoubleVotingEvidence(ctx, *evidence, chainID, consuPubkey); err != nil {
+		return err
+	}
+
+	// execute the jailing and tombstoning
+	k.JailAndTombstoneValidator(ctx, providerAddr)
+
+	k.Logger(ctx).Info(
+		"confirmed equivocation",
+		"byzantine validator address", providerAddr,
+	)
+
+	return nil
+}
+
+// VerifyDoubleVotingEvidence verifies a double voting evidence
+// for a given chain id and validator public key
+func (k Keeper) VerifyDoubleVotingEvidence(
+	ctx sdk.Context,
+	evidence tmtypes.DuplicateVoteEvidence,
+	chainID string,
+	pubkey tmprotocrypto.PublicKey,
+) error {
+	// TODO: check the evidence age once we agreed on the infraction height mapping
```

Comment:
Use only the timestamp (see https://github.com/cosmos/cosmos-sdk/blob/5b02261cb5d52c1bb645efa559195a4e5b0a7945/x/evidence/keeper/infraction.go#L58), i.e.,

```
infractionTime := evidence.GetTime()
ageDuration := ctx.BlockHeader().Time.Sub(infractionTime)
// Reject evidence if the double-sign is too old. Evidence is considered stale
// if the difference in time is greater than the allowed parameters defined.
cp := ctx.ConsensusParams()
if cp != nil && cp.Evidence != nil {
    if ageDuration > cp.Evidence.MaxAgeDuration{
        ...
```

Reply:
Done.